nexus-tweaks.user.js

```diff
@@ -32,6 +32,414 @@ function timesCharExist(x, c){
+      resultHtml = resultHtml
+          + '<div class="char-div-element bar-segments-div" style="left: '
+          + (188 + hpseg * scalingFactor) + 'px;"></div>'
+    }
+    resultHtml = resultHtml + '</div>';
+    return resultHtml;
+  }
+
+  const characterList = (mod) => {
+    const coordArea = document.getElementById('AreaDescription');
+    const localStoragePrefix = "ntcl-";
+    const localStorageLastLocationKey = localStoragePrefix + 'last-location'
+    const localStorageHpPrefix = localStoragePrefix + 'person-hp-'
+    if (!coordArea) {
+      // dead
+      clearModuleLocalStorage(localStoragePrefix);
+      return;
+    }
+    let charArea = document.getElementsByClassName('charListArea')[0];
+    let peopleMatch = /There (is|are) (\d+) other (person|people) here, (.*)/
+    .exec(charArea.innerHTML);
+
+    let locationMatch = /(\d+, \d+ .+?),/.exec(coordArea.innerHTML);
+    let location = 'n/a'
+    // do not fail when location is not available
+    if (locationMatch && locationMatch[1]) {
+      location  = locationMatch[1];
+    }
+    let previousLocation = localStorage.getItem(localStorageLastLocationKey);
+    let locationUnchanged = (location === previousLocation && location !== 'n/a');
+    localStorage.setItem(localStorageLastLocationKey, location);
+
+    if (!peopleMatch || (peopleMatch[2] === '0')) {
+      // no people present, can clear saved previous HP values
+      clearModuleLocalStorage(localStoragePrefix);
+      localStorage.setItem(localStorageLastLocationKey, location);
+      return;
+    }
+    let charHtmls = peopleMatch[4].substring(1, peopleMatch[4].length - 1).split(
+        '>, <');
+    let charCount = charHtmls.length;
+    let personList = [];
+
+    // save list of people previously seen on last refresh.
+    // used to remove entries for people no longer present later
+    const peopleToDelete = Object.keys(localStorage)
+      .filter(key => key.indexOf(localStorageHpPrefix) === 0);
+
+    // create list of people and calculate sorting value
+    for (let i = 0; i < charCount; i++) {
+      let element = charHtmls[i];
+      let person = createPerson(element);
+      calculateSortValue(person);
+      updatePersonFromLocalStorage(person, localStorageHpPrefix, locationUnchanged);
+      personList.push(person);
+
+      // remove from list of people marked for deletion from local storage
+      const index = peopleToDelete.indexOf(localStorageHpPrefix + person.id);
+      if (index > -1) {
+        peopleToDelete.splice(index, 1);
+      }
+    }
+
+    // remove all previously saved people that are no longer present
+    // so they can register as new again when they reappear
+    for(const personToDelete of peopleToDelete) {
+      localStorage.removeItem(personToDelete);
+    }
+
+    // sort list of people, higher sortValue sorts to top of list
+    personList = personList.sort((a, b) => b.sortValue - a.sortValue);
+
+    let title = 'There ' + peopleMatch[1] + ' ' + personList.length
```

Comment:
nit: Another great place for template strings.

Reply:
done!